lib/pure/strformat.nim

```diff
@@ -573,15 +573,14 @@ template formatValue(result: var string; value: char; specifier: string) =
 template formatValue(result: var string; value: cstring; specifier: string) =
   result.add value
 
-proc strformatImpl(pattern: NimNode; openChar, closeChar: char): NimNode =
-  if pattern.kind notin {nnkStrLit..nnkTripleStrLit}:
-    error "string formatting (fmt(), &) only works with string literals", pattern
+proc strformatImpl(f: string; openChar, closeChar: char): NimNode =
+  let info = callsite()
+  # pending https://github.com/nim-lang/RFCs/issues/387, use `callTree`
```

Comment:
Can you find a way to do this for now without callsite and without RFC 387 ?

Reply:
last commit removes callsite (but I still think RFC 387 is a good thing to implement regardless), here's the semantic difference:
```nim
when defined case1:
  import std/strformat
  let x = @[3, 4]
  let a = fmt"foo {nonexistent}"

when defined case2:
  import std/strformat
  let x = @[3, 4]
  let a = fmt"foo {x * }"
```
case1: no difference
case2:
before PR:
```
stack trace: (most recent call last)
/Users/timothee/git_clone/nim/Nim_devel/lib/pure/strformat.nim(673, 53) fmt
/Users/timothee/git_clone/nim/Nim_devel/lib/pure/strformat.nim(639, 18) strformatImpl
/Users/timothee/git_clone/nim/timn/tests/nim/all/t12419.nim(5, 1) template/generic instantiation from here
/Users/timothee/git_clone/nim/timn/tests/nim/all/t12419.nim(13, 14) Error: could not parse `x * `.
/Users/timothee/git_clone/nim/Nim_devel/lib/core/macros.nim(530, 5) Error: expression expected, but found '[EOF]'
    let a = fmt"foo {x * }"
```

after PR (after the commit that removes callsite):
```
stack trace: (most recent call last)
/Users/timothee/git_clone/nim/Nim_prs/lib/pure/strformat.nim(680, 16) fmt
/Users/timothee/git_clone/nim/Nim_prs/lib/pure/strformat.nim(634, 16) strformatImpl
/Users/timothee/git_clone/nim/timn/tests/nim/all/t12419.nim(5, 1) template/generic instantiation from here
/Users/timothee/git_clone/nim/timn/tests/nim/all/t12419.nim(13, 14) template/generic instantiation of `fmt` from here
/Users/timothee/git_clone/nim/Nim_prs/lib/pure/strformat.nim(634, 16) Error: could not parse `x * ` in `foo {x * }`.
/Users/timothee/git_clone/nim/Nim_prs/lib/core/macros.nim(531, 5) Error: expression expected, but found '[EOF]'
            error("could not parse `$#` in `$#`.\n$#" % [subexpr, f, e.msg])
```

i'd argue it's better after PR because it shows more context in stacktrace and all the information is still there.